service/src/main/frontend/src/functions/formatters.ts

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-var */
+import { Ref, WritableComputedRef, computed, defineComponent, ref } from 'vue';
+
+// Parses a Value to remove uncessessary 'undefined' or empty values.
+export function parseVal(colName: any, colValue: any): string {
+  if (colName === 'Types' || colName === 'Descriptions') {
+    if (colValue == undefined) {
+      return '';
+    } else {
+      return colValue.toString();
+    }
+  } else {
+    return colValue;
+  }
+}
+
+// Produces the max substring for the table, ads '...' if above 34 chars.
+export function maxSubstring(str: any, colName: any): any {
+  if (str == undefined) {
+    return;
+  } else if ((colName === 'fieldName' || colName === 'internalFieldName') && str.length > 32) {
+    return str.substring(0, 30) + ' ...';
+  } else if ((colName === 'Types') && str.length > 14) {
+    return str.substring(0, 9) + ' ...';
```

Comment:
Just curious, why are we subtracting 5 from the length here instead of 2 in the other cases?

Reply:
Because the  'Types' column is different in terms of length. After testing, if I did 2 it would overlap with the 'Tokenized'.

Reply:
Ok that makes sense, could you add a comment saying that?

Reply:
Will do